addons/l10n_fr_hr_holidays/models/hr_leave.py

```diff
@@ -34,26 +32,25 @@ def _get_fr_date_from_to(self, date_from, date_to):
             raise UserError(_("An employee can't take paid time off in a period without any work hours."))
 
         if not self.request_unit_hours:
-            # extend the date range to the full day or requested period so that the difference of hours
-            # between employee's work hours and the company's work hours doesn't affect the computation
-            date_from = date_from.replace(hour=0, minute=0, second=0)
-            date_to = date_to.replace(hour=23, minute=59, second=59)
+            # Use company's working schedule hours for the leave to avoid duration calculation issues.
 
             def adjust_date_range(date_from, date_to, period, attendance_ids, employee_id):
-                period_ids = attendance_ids.filtered(lambda a: a.day_period == period)
-                max_hour = max(attendance.hour_to for attendance in period_ids)
-                min_hour = min(attendance.hour_from for attendance in period_ids)
-                date_from = self._to_utc(date_from, min_hour, employee_id)
-                date_to = self._to_utc(date_to, max_hour, employee_id)
+                period_ids_from = attendance_ids.filtered(lambda a: a.day_period in period and int(a.dayofweek) == date_from.weekday())
+                period_ids_to = attendance_ids.filtered(lambda a: a.day_period in period and int(a.dayofweek) == date_to.weekday())
+                if period_ids_from:
+                    min_hour = min(attendance.hour_from for attendance in period_ids_to)
+                    date_from = self._to_utc(date_from, min_hour, employee_id)
+                if period_ids_to:
+                    max_hour = max(attendance.hour_to for attendance in period_ids_from)
+                    date_to = self._to_utc(date_to, max_hour, employee_id)
                 return date_from, date_to
 
             if self.request_unit_half:
-                attendance_ids = self.company_id.resource_calendar_id.attendance_ids.filtered(
-                    lambda a: int(a.dayofweek) == date_to.weekday() and a.day_period != "lunch"
-                )
-                if attendance_ids:
-                    period = 'morning' if self.request_date_from_period == 'am' else 'afternoon'
-                    date_from, date_to = adjust_date_range(date_from, date_to, period, attendance_ids, self.employee_id)
+                period = 'morning' if self.request_date_from_period == 'am' else 'afternoon'
```

Comment:
If you don't set the string inside a list, it will break when given to `adjust_date_range`
```suggestion
                period = ['morning' if self.request_date_from_period == 'am' else 'afternoon']
```
or
```suggestion
                period = ['morning'] if self.request_date_from_period == 'am' else ['afternoon']
```